src/Middleware/ErrorOverlayMiddleware.js

```diff
@@ -0,0 +1,145 @@
+const ErrorStackParser = require("error-stack-parser");
+const debug = require("debug")("Eleventy:ErrorOverlayMiddleware");
+
+/**
+ * A browser-sync middleware that pipes build errors to a browser view regardless of the requested
+ * path.
+ *
+ * @param {() => Error | null} getError a callback evaluated on every request that should return an
+ *  error to be shown (if any). Returning `null` bypasses the middleware.
+ *
+ * @returns {Function} the middleware
+ */
+module.exports = function ErrorOverlayMiddleware(getError) {
+  return function(_req, res, next) {
+    const error = getError();
+
+    if (!error) {
+      next();
+      return;
+    }
+
+    debug("Showing error", error);
+    const stacktrace = ErrorStackParser.parse(error).map(item => ({
+      ...item,
+      fileName: item.fileName
+        .replace(process.cwd(), ".")
+        .replace(process.env.HOME, "~")
+    }));
+
+    res.status = 500;
+    res.setHeader("content-type", "text/html");
+    res.end(
+      template({
+        errorName: error.name,
+        errorMessage: error.message,
+        stacktrace
+      })
+    );
+  };
+};
+
+const template = ({ errorName, errorMessage, stacktrace }) => `
```

Comment:
I chose to implement this as a JS template to avoid pulling in `nunjucks` just to render one small template. I also didn't know where I would even put a nunjucks file in this hierarchy. Feedback welcome here :)